ext/lib/crypto/mbedtls/CMakeLists.txt

```diff
@@ -112,6 +114,12 @@ else()
   # Lib mbedtls_external depends on libgcc (I assume?) so to allow
   # mbedtls_external to link with gcc we need to ensure it is placed
   # after mbedtls_external on the linkers command line.
+
+  target_link_libraries(mbedTLS INTERFACE zephyr_interface)
+
+else()
+
+  assert(CONFIG_MBEDTLS_EXTERNAL_CONFIG "MBEDTLS was enabled, but neither BUILTIN, LIBRARY or EXTERNAL_CONFIG was selected.")
```

Comment:
I'm confused about how this works.

ext/lib/crypto/mbedtls/CMakeLists.txt

will only be executed if CONFIG_MBEDTLS due to

https://github.com/zephyrproject-rtos/zephyr/blob/7f4bc0dc40277bd9afbf59a66c0c2ac6c58f06ab/ext/lib/crypto/CMakeLists.txt#L2

But CONFIG_MBEDTLS will never be set when MBEDTLS_EXTERNAL_CONFIG is set due to https://github.com/zephyrproject-rtos/zephyr/pull/9984/commits/7f4bc0dc40277bd9afbf59a66c0c2ac6c58f06ab#diff-a4eeb121fd0284079c8218ec6105fdc6R26

so this assert will never occur.

Perhaps either we should have

elseif(CONFIG_MBEDTLS_EXTERNAL_CONFIG)
assert(0 "Unreachable code")
endif()

to indicate to the reader of this cmake code that this condition is not possible. (Right now it reads as if the user has misconfigured something, but this is not possible).

Reply:
Before this PR assert was shown in following situation:
```
MBEDTLS = y
MBEDTLS_BUILTIN = n
MBEDTLS_LIBRARY = n
```
I added `MBEDTLS_EXTERNAL_CONFIG`, so I had to add it to this assert, so it will be show in following situation:
```
MBEDTLS_EXTERNAL_CONFIG = n
MBEDTLS = y
MBEDTLS_BUILTIN = n
MBEDTLS_LIBRARY = n
```
It is possible and informs that the user has misconfigured something.

Reply:
Right, but then the error message should be saying:

"MBEDTLS was enabled, but neither BUILTIN or LIBRARY was selected."

Because you can't have

MBEDTLS=y
MBEDTLS_EXTERNAL_CONFIG=y

which is what the message is implying that you can have.

Alternatively the more explicit message,

"MBEDTLS was enabled, but neither BUILTIN or LIBRARY was selected. Either disable MBEDTLS and enable EXTERNAL_CONFIG. Or enable either BUILTIN or LIBRARY."

Reply:
OK, I will change the text message.